### PR TITLE
Update test case content

### PR DESCRIPTION
**Pull request by Brrovko**

### CODE REVIEW

**Review by Georgiy-Petrov-Quantori — APPROVED** [review]

**Review by SergeyRuzin — APPROVED** [review]

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs [hunk 1]
```diff
@@ -102,5 +91,26 @@ public IEnumerable<Case> GetCases(ulong projectId, ulong suiteId)
             ValidateRequestResult(result);
             return result.Payload;
         }
+
+        private static bool CompareTestCaseContent(CreateCaseRequest createCaseRequest, Case testRailCase)
+        {
+            if(!testRailCase.Title.Equals(createCaseRequest.Title)) return true;
+            if(!testRailCase.TemplateId.Equals(createCaseRequest.TemplateId)) return true;
```
Comment (vladimir-s-quantory) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs:98]: One return statement should be enough, I think:
`return (!testRailCase.Title.Equals(createCaseRequest.Title) || !testRailCase.TemplateId.Equals(createCaseRequest.TemplateId));`
Also, the name of this method suggests that we should return some results of this comparison - in our case it should be something like **CheckTestCaseEquality**
And can you please explain why Titles and TemplateIds of compared cases should not be equal to each other?

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/TestRailSynchronizer.cs [hunk 2]
```diff
@@ -37,23 +38,21 @@ public void Sync(List<IFeatureFile> featureFiles)
                 {
                     var tagId = scenario.Tags.FirstOrDefault(tag => Regex.Match(tag.Name, config.TagIdPattern, RegexOptions.IgnoreCase).Success);
 
+                    var createCaseRequest = _caseContentBuilder.BuildCreateCaseRequest(scenario, featureFile);
```
Comment (vladimir-s-quantory) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/TestRailSynchronizer.cs:41]: I am not sure that it is correct naming policy - I propose to rename the variable to **caseRequest** and method to **BuildCreateOrUpdateCaseRequest**

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs (hunk 1 above)
Comment (Brrovko) [reply GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Client/TestRailClientWrapper.cs]: image
In this case, the second condition and a return statement are missed.

Naming fixed.

Titles and TemplateIds just checked for equality.

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/TestRailSynchronizer.cs (hunk 2 above)
Comment (Brrovko) [reply GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/TestRailSynchronizer.cs]: Fixed

**Review by vladimir-s-quantory — DISMISSED** [review]

**Review by Georgiy-Petrov-Quantori — APPROVED** [review]

**Review by Georgiy-Petrov-Quantori — APPROVED** [review]

**Review by SergeyRuzin — APPROVED** [review]

**Pull request merged**